### PR TITLE
Mpi::is_probably_prime (mbedtls_mpi_is_prime_ext)

### PR DESCRIPTION
**Pull request by zugzwang**
Support mbedtls_mpi_is_prime, the Miller-Rabin primality test.

### CODE REVIEW

File: mbedtls/src/bignum/mod.rs [hunk 1]
```diff
@@ -407,6 +408,21 @@ impl Mpi {
 
         Ok(r)
     }
+
+    /// The Miller-Rabin primality test. Returns `MpiNotAcceptable` is self is
```
Comment (raoulstrackx) [inline mbedtls/src/bignum/mod.rs:412]: Typo: "_if_ self is"

File: mbedtls/src/bignum/mod.rs [hunk 2]
```diff
@@ -407,6 +408,21 @@ impl Mpi {
...
+    /// definitely not prime. If no error is returned, self is probably prime. See
+    /// mbedtls_mpi_is_prime.
+    pub fn is_prime<F: Random>(&self, rng: &mut F) -> Result<()> {
+        unsafe {
+            mpi_is_prime(
```
Comment (raoulstrackx) [inline mbedtls/src/bignum/mod.rs:417]: The `mbedtls_mpi_is_prime` function is deprecated in favor of `mbedtls_mpi_is_prime_ext` that requires specifying the number of rounds of the Miller-Rabin test. I'm wondering if we should provide the same functionality.

Comment (zugzwang) [reply mbedtls/src/bignum/mod.rs:417]: I missed that, thanks for pointing this out. Yes, definitely.

File: mbedtls/src/bignum/mod.rs [hunk 3]
```diff
@@ -407,6 +408,23 @@ impl Mpi {
 
         Ok(r)
     }
+
+    /// The Miller-Rabin primality test with k rounds. Returns
+    /// `MpiNotAcceptable` if self is definitely not prime. If no error is
+    /// returned, self is prime with a probability of 1 - 1/2^{2k}. See
+    /// mbedtls_mpi_is_prime.
+    pub fn is_prime<F: Random>(&self, k: u32, rng: &mut F) -> Result<()> {
```
Comment (jethrogb) [inline mbedtls/src/bignum/mod.rs:416]: Should be `is_probably_prime`?

**Pull request merged**